### PR TITLE
Added support of DBAllocator service in middleware provider tests.

### PR DESCRIPTION
**Pull request by hhovsepy**
Added new methods to allocate and deallocate database on request.

Usage:
properties = db_allocate('oracle12c')
db_deallocate(properties['uuid'])

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit fb09f5fdf280ba348447f4a2ca1b5a6f2bb333db:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b4228a0351eefcacdd7f0056350af42a85ac8b1d:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/dballocator_methods.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            hibernate.dialect: org.hibernate.dialect.{}Dialect
+            hibernate.connection.url: formatted jdbc url to connect
+            hibernate.connection.schema: schema name
+
+    Usage:
+        properties = db_allocate('oracle12c')
+    """
+    try:
+        db_allocate_url = conf.cfme_data\
+            .get('resources', {})['databases']['db_allocator'][DBALLOCATOR_URL_KEY]
```
Comment (psav) [inline cfme/tests/middleware/dballocator_methods.py]: why make the ```DBALLOCATOR_URL_KEY``` a const here?

Comment (hhovsepy) [inline cfme/tests/middleware/dballocator_methods.py]: or make all keys as constants? to reuse them, java style

**Comment by dajoRH** [comment]
Lint report for commit 081ed906ddf6291e60cf6d2d0d85ca383fe4d203:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/dballocator_methods.py (hunk 1 above)
Comment (hhovsepy) [inline cfme/tests/middleware/dballocator_methods.py]: @psav changed and rebased

File: cfme/tests/middleware/dballocator_methods.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            hibernate.dialect: org.hibernate.dialect.{}Dialect
+            hibernate.connection.url: formatted jdbc url to connect
+            hibernate.connection.schema: schema name
+
+    Usage:
+        properties = db_allocate('oracle12c')
+    """
+    try:
+        db_allocate_url = conf.cfme_data\
+            .get('resources', {})['databases']['db_allocator']['db_allocate_url']
```
Comment (psav) [inline cfme/tests/middleware/dballocator_methods.py]: We need to make this more robust and ensure this data is in the yamls

Comment (psav) [inline cfme/tests/middleware/dballocator_methods.py]: Also
```python
        db_allocate_url = conf.cfme_data.get(
            'resources', {})['databases']['db_allocator']['db_allocate_url']
```

**Comment by dajoRH** [comment]
Lint report for commit 1f3f5b10739cdd25df32d7e132672057f5196284:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 323cc821d339e435cdf7f104dca9d76a917f6c3a:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/dballocator_methods.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            db.hostname: database hostname
+            datasource.class.xa: XADataSource class
+            server_uid: server uid
+            hibernate33.dialect: org.hibernate.dialect{}Dialect
+            server_labels: server_label
+            hibernate.connection.username: username
+            db.jdbc_class: JDBC Driver Class
+            db.schema: schema name
+            hibernate.connection.driver_class: Driver Class
+            uuid=2f7afb52-d4ca-4c33-9017-b7f12bfc4c70
```
Comment (RonnyPfannschmidt) [inline cfme/tests/middleware/dballocator_methods.py]: typo

File: cfme/tests/middleware/dballocator_methods.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    Args:
+        db_type: type of the database which is registered in DBAllocator,
+            all possible types are in this class.
+        expiry: reservation duration in minutes after
+            which database will be automatically deallocated (optional).
+
+    Returns:
+        database connection properties as dictionary containing information:
+            db.password: password
```
Comment (RonnyPfannschmidt) [inline cfme/tests/middleware/dballocator_methods.py]: given the dotted/nested structure i wonder if we shouldn't have a simple object representing the data and having the deallocate method (in order to localize information and remove the need to know the structure on use)

File: cfme/tests/middleware/dballocator_methods.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    Args:
+        uuid: uuid of database instance in DBAllocator tool.
+
+    Usage:
+        properties = db_allocate('oracle12c')
+        db_deallocate(properties['uuid'])
+    """
+    try:
+        db_deallocate_url = conf.cfme_data.get(
```
Comment (RonnyPfannschmidt) [inline cfme/tests/middleware/dballocator_methods.py]: this line will fail on missing data and should not be in the try/except for a different error

File: cfme/tests/middleware/test_middleware_datasource.py [hunk 6]
```diff
@@ -14,6 +16,13 @@
 ITEMS_LIMIT = 2  # when we have big list, limit number of items to test
 
 
+@pytest.yield_fixture(scope="function")
+def oracle_database():
```
Comment (RonnyPfannschmidt) [inline cfme/tests/middleware/test_middleware_datasource.py]: if there is always a need to ensure allocation/deallocation,

how about making that db_allocate a context-manager to begin with

```
with allocated_db(ORACLE_12C) as db_metadata:
  yield db_metadata
```
seems like a potential candidate

File: cfme/tests/middleware/dballocator_methods.py (hunk 4 above)
Comment (hhovsepy) [inline cfme/tests/middleware/dballocator_methods.py]: my bad, it is actually dictionary with '=', so fixing the comments

**Comment by dajoRH** [comment]
Lint report for commit 3e3837b89a4f1849fa2eb1d92429167a41822d1d:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/test_middleware_datasource.py (hunk 6 above)
Comment (hhovsepy) [inline cfme/tests/middleware/test_middleware_datasource.py]: what is wrong here in using fixture as a setup/teardown for database resource? @psav wdyt?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c4099b08d14cbd336d6d59ba65d4008da4dbe40a:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9ea60bf71df004b35384066d08048ba33483d7ec:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**